test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +94,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(commandUsed.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
```

Comment:
also as discussed during tutorial, we shouldn't assert that they're equal only if they're mutating.

We should also make sure that commands, that are not mutating, aren't affecting the save file too.